src/Compilers/Core/Portable/InternalUtilities/ReflectionUtilities.cs

```diff
@@ -101,32 +126,60 @@ internal static ConstructorInfo GetDeclaredConstructor(this TypeInfo typeInfo, p
             return FindItem(typeInfo.DeclaredConstructors, paramTypes);
         }
 
+#if USES_ANNOTATIONS
+        public static T CreateDelegate<T>(this MethodInfo? methodInfo)
+#else
         public static T CreateDelegate<T>(this MethodInfo methodInfo)
+#endif
         {
             if (methodInfo == null)
             {
+#if USES_ANNOTATIONS
+                // PROTOTYPE(NullableDogfood): We will fix with System.Delegate constraint
```

Comment:
💭 Isn't this constraint already enabled in this branch due to the use of C# 8.0?

Reply:
The compiler with nullable implementation is based off 15.6 at the moment. `System.Delegate` was implemented in 15.7.